The Quartermaster Hardsuit

Endecc opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
<!-- What did you change in this PR? -->
Added the ~Quartermaster~ luxury mining hardsuit, a requested (and debated) hardsuit specially made for a Quartermaster*.
*ᴺᵒᵗ ˢᵖᵉᶜᶦᶠᶦᶜᵃˡˡʸ ᶠᵒʳ ᵗʰᵉ ᵠᵘᵃʳᵗᵉʳᵐᵃˢᵗᵉʳ.
Does **not** add to any roundstart lockers.
Added a new cargo order, the ~Quartermaster's~ luxury mining hardsuit, at 15,000 Spesos.

## Why / Balance
<!-- Why was it changed? Link any discussions or issues here. Please discuss how this would affect game balance. -->
_The following was for when it was labelled as the Quartermaster's specifically. I can't be bothered to reedit it all._
- **Quartermaster is a desk job, why a hardsuit?** - People have now and again asked for a QM hardsuit, but it's hotly debated because their job doesn't really need it. Salvagers have hardsuits a QM could steal, they have EVA access themselves, or they could outright buy an EVA package themselves. These are all valid points against a roundstart hardsuit, and I agree it's not really necessary in such an implementation. That's why I went a _different_ route.
- **Order it, just to say you could** - The Quartermaster does not start with this hardsuit. Instead, it must be ordered for 15,000 spesos, as a flaunt of department wealth. It's not a particularly good hardsuit, it's slightly faster than its underling Mining hardsuit, but offers roughly a THIRD of the protection
- **It's all in character** - Cargo's whole internal shtick is to make a lot of money, and flaunt it. But there's never really been one good goal item specifically for cargo. Centcomm has heard the pleas of Quartermasters around their sectors, and they have finally listened. Of course, they aren't going to pass up the opportunity to siphon some funds back from you. They know you're good for it.

## Technical details
<!-- If this is a code change, summarize at high level how your new code works. This makes it easier to review. -->

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
_Renamed to "luxury mining hardsuit", sprites remain the same_
![qmhardsuit](https://github.com/space-wizards/space-station-14/assets/33369477/1b1d7d2d-93ef-4707-ae7b-d5c3ca5538dc)
![qmhardsuitcargo](https://github.com/space-wizards/space-station-14/assets/33369477/bbe88c14-dfd5-4b9e-b6f9-2eb098c78be0)

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->

:cl:
- add: Added the luxury mining hardsuit, a lighter and more versatile mining hardsuit styled after the Quartermaster.
- add: Added luxury mining hardsuit cargo order, because this ELITE cargo suit is too expensive to stock on every station.

lzk228 commented:
QM don't need it
all their work on station, there's no need to go in space, and even if necessary, there is salvages suits and EVA

Endecc commented:
> QM don't need it all their work on station, there's no need to go in space, and even if necessary, there is salvages suits and EVA

I specifically addressed these concerns, _point for point_, in the Why / Balance section. It is an obscenely expensive, single hardsuit that does not offer any exceptional protections. They do not _need_ it, no.
However, it is funny. That is why I implemented the way I have as cargo order. Again, it is **not a roundstart hardsuit.**

Flareguy commented:
The sprite for this is so dumb. I love it.

EmoGarbage404 commented:
If you want a big ticket item to flaunt just port the bicycle crate from /tg/

BolloTea commented:
In my opinion this is a very stupid idea.  Like a quartermaster suit, like the way you buy it.  With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

Aisu9 commented:
> In my opinion this is a very stupid idea. Like a quartermaster suit, like the way you buy it. With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

another reason to make the request console's catalogue access/id dependant

Endecc commented:
> If you want a big ticket item to flaunt just port the bicycle crate from /tg/

Back when I made a push to replicate the Advanced Mop function from SS13, I was told that making something new is better than just copying the same stuff from SS13. To my knowledge, this is fulfilling that preference.

> In my opinion this is a very stupid idea. Like a quartermaster suit, like the way you buy it. With us, everyone who has access to the console in the cargo will be able to buy his suit... If you really want to add it to the game (which many really consider it useless), then you need to find another way to get it.

Compared to the other Head of Department hardsuits, it's really nothing special. If the cargo department seriously wants to shelve the tens of thousands it would take to get multiple of them, why not?

Flareguy commented:
You should probably leave a comment in the QM locker prototype and on/or on the hardsuit prototype telling people to not put it in the locker, or else someone might end up doing it.

mirrorcult commented:
do not map suffix probably necessary

mirrorcult commented:
no problem with this conceptually, its good to have stuff that costs a ton and its funny when its mostly just a status symbol

Endecc commented:
> do not map suffix probably necessary

Sounds good to me. Added comments to the hardsuit, locker, and helmet.

potato1234x commented:
i love this

EmoGarbage404 commented:
I still think that this being specifically identified as QM's hardsuit is weird. The concept of a luxury hardsuit is a funny concept for a cargo crate but I think specifically identifying it with the QM is strange. It not only makes it the only head hardsuit to not spawn with the head, but also the only one that has multiple instances and can be easily acquired by any crew member. I think players may find it confusing since it seems to violate the norms that the other head hardsuits establish.

Endecc commented:
> I still think that this being specifically identified as QM's hardsuit is weird. The concept of a luxury hardsuit is a funny concept for a cargo crate but I think specifically identifying it with the QM is strange. It not only makes it the only head hardsuit to not spawn with the head, but also the only one that has multiple instances and can be easily acquired by any crew member. I think players may find it confusing since it seems to violate the norms that the other head hardsuits establish.

While the order of a general luxury hardsuit is fair, I don't personally see it as funny as Centcomm specifically acknowledging the desires of QMs wanting their own hardsuit, and making them order a specially-designed one at an exorbitant rate just for the privilege of saying they have their own. To me, that's more interesting than something like "This hardsuit is really expensive because it's made out of GOLD", or what-have-you.

The differences from other head hardsuits by having it ordered is very intentional, in that way. It IS violating the norms, because people don't want it to be part of the norm at all, it'd be an unnecessary shift-start boon for a desk job. It's a very intentional departure explicitly to get people to stop asking for a QM hardsuit, because Centcomm doesn't want to hear it anymore. They're getting what they wanted, just not in the most conventional way.

Endecc commented:
Feedback discussion in discord #contributors agreed that it would be better renamed, so I've done that now with some minor tweaks to the descriptions so it makes sense.

LankLTE commented on Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -390,6 +390,36 @@
   - type: ToggleableClothing
     clothingPrototype: ClothingHeadHelmetHardsuitSecurityRed
 
+#Quartermaster's Hardsuit
+- type: entity
+  parent: ClothingOuterHardsuitBase
+  id: ClothingOuterHardsuitQm #DO NOT MAP - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
+  name: luxury mining hardsuit
+  description: A refurbished mining hardsuit, fashioned after the Quartermaster. Graphene lining provides less protection, but is much easier to move.
```
You need to remove references (ID, comment, description) to this being related to the QM. Ideally the sprite would be renamed too.

Endecc replied:
I was told leaving description references to the Quartermaster would be fine. I can definitely change the internal thongs for sure, but is it seriously imperative to remove every vein and spine of intent I had with the original PR?

LankLTE replied:
Just looked through the relevant discord discussion. Description is fine, yeah.

LankLTE commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -300,6 +300,25 @@
         Piercing: 0.9
         Heat: 0.9
 
+#Quartermaster's Hardsuit
+- type: entity
+  parent: ClothingHeadHardsuitWithLightBase
+  id: ClothingHeadHelmetHardsuitQm #DO NOT MAP - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
```
Same as above

LankLTE commented on Resources/Prototypes/Catalog/Cargo/cargo_cargo.yml:
```diff
@@ -6,4 +6,14 @@
+  group: market
+
+- type: cargoProduct
+  id: CargoQuartermasterHardsuit
+  icon:
+    sprite: Clothing/Head/Hardsuits/qm.rsi
+    state: icon
+  product: CrateCargoQuartermasterHardsuit
```
^

LankLTE commented on Resources/Locale/en-US/prototypes/catalog/cargo/cargo-cargo.ftl:
```diff
@@ -0,0 +1,2 @@
+ent-CargoQuartermasterHardsuit = { ent-CrateCargoQuartermasterHardsuit }
+    .desc = { ent-CrateCargoQuartermasterHardsuit.desc }
```
^

LankLTE commented on Resources/Locale/en-US/prototypes/catalog/fills/crates/cargo-crates.ftl:
```diff
@@ -0,0 +1,2 @@
+ent-CrateCargoQuartermasterHardsuit = luxury mining hardsuit crate
+    .desc = Finally, a hardsuit Quartermasters could call their own. Centcomm has heard you, now stop asking.
```
Remove the reference to this being for the QM.

Endecc replied:
Same as hardsuits.yml comment - removing the spine of the PR

LankLTE replied:
This is similarly fine.

LankLTE commented on Resources/Prototypes/Catalog/Fills/Crates/cargo.yml:
```diff
@@ -0,0 +1,7 @@
+- type: entity
+  id: CrateCargoQuartermasterHardsuit
+  parent: CrateCommandSecure
+  components:
+  - type: StorageFill
+    contents:
+    - id: ClothingOuterHardsuitQm
```
As above

LankLTE commented on Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -5,6 +5,7 @@
   components:
   - type: StorageFill
     contents:
+    #DO NOT MAP QUARTERMASTER HARDSUIT LOCKER - https://github.com/space-wizards/space-station-14/pull/19738#issuecomment-1703486738
```
This can be removed too

LankLTE reviewed (changes requested):
Hardsuit needs to internally not be the QMs' too- don't just change the name of the suit.

EtereohZ commented:
Should a "Luxury" category be added to the cargo console if more premium/luxury items are, maybe, going to be added in the future?

Endecc commented:
> Should a "Luxury" category be added to the cargo console if more premium/luxury items are, maybe, going to be added in the future?

The cargo category the hardsuit is in is extremely bare-bones already, serving only 1 order for an extra cargo palette. If such a category is made, it wouldn't be just for this. That'd make two categories with 1 item.

EmoGarbage404 commented:
Fix the changelog to reflect the PR + be less wordy

Endecc commented:
> Fix the changelog to reflect the PR + be less wordy

done
